eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -79,24 +116,42 @@ private SubmittedVariantEntity getSubmittedVariantEntity(Document notClusteredSu
 
     @Override
     public void open(ExecutionContext executionContext) throws ItemStreamException {
+        if (executionContext.containsKey(CURRENT_ID_KEY)) {
+            this.currentId = executionContext.getString(CURRENT_ID_KEY);
+            this.currentCollection = executionContext.getString(CURRENT_COLLECTION_KEY);
+        }
+        initializeRetryTemplate();
         initializeReader();
     }
 
+    public void initializeRetryTemplate() {
+        retryTemplate = new RetryTemplate();
+        retryTemplate.setRetryPolicy(new MongoReaderRetryPolicy());
+        retryTemplate.setBackOffPolicy(new ExponentialRandomBackOffPolicy());
+    }
+
     public void initializeReader() {
         Bson query = Filters.and(Filters.in(ASSEMBLY_FIELD, assembly),
                                  Filters.exists(CLUSTERED_VARIANT_ACCESSION_FIELD, readOnlyClusteredVariants));
+        Bson queryWithCurrentId = null;
+        if (currentId != null) {
+            queryWithCurrentId = Filters.and(query, Filters.gt(ID_FIELD, currentId));
+            logger.info("Issuing find in {} collection: {}", currentCollection, queryWithCurrentId);
+        }
         logger.info("Issuing find: {}", query);
```

Comment:
When you retry you would log twice. Maybe doing it only once with a generic log is better?

Reply:
It's a bit confusing actually, when we retry we might actually be making two different queries - for example, if `readOnlyClusteredVariants` is true and the failure occurred while reading the dbSNP collection, we'll query the dbSNP collection with the currentId filter but the EVA collection with the standard query.

I was being lazy about the logging but I'll change it to be more specific as to what we're querying in which collection, because I can definitely see us getting confused by it in the (near) future...